### PR TITLE
Verbose logging cleanup

### PR DESCRIPTION
**Pull request by robertwb**
This should suppress some sources of (severe-looking) logging noise that I've noticed recently.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

File: sdks/python/apache_beam/runners/portability/fn_api_runner_test.py [hunk 1]
```diff
@@ -26,6 +27,8 @@
 import unittest
 from builtins import range
 
+import six
```
Comment (udim) [inline sdks/python/apache_beam/runners/portability/fn_api_runner_test.py]: We don't use `import six` any more in Beam (was systematically removed but I don't remember the reasoning). @tvalentyn might know more.

File: sdks/python/apache_beam/runners/portability/fn_api_runner_test.py [hunk 2]
```diff
@@ -47,11 +50,31 @@
   DEFAULT_SAMPLING_PERIOD_MS = 0
 
 
+def suppress_stderr(f):
```
Comment (udim) [inline sdks/python/apache_beam/runners/portability/fn_api_runner_test.py]: Naming nit:
Following nose's naming conventions (--nocapture, --nologcapture), I'd call this wrapper `capture_stderr`.

Comment (udim) [reply sdks/python/apache_beam/runners/portability/fn_api_runner_test.py]: Also, we could alternatively remove --nocapture, --nologcapture flags from unit tests, which should have the same effect.

Comment (robertwb) [reply sdks/python/apache_beam/runners/portability/fn_api_runner_test.py]: Changed. I also rebased due to merge conflicts. 

Having nocapture on in general is quite useful for tests that hang. I looked around and didn't see a way to control it programatically per test.

**Comment by udim** [comment]
previous precommit failed in an odd way: https://builds.apache.org/job/beam_PreCommit_Python_Commit/3576/console

File: sdks/python/apache_beam/runners/portability/fn_api_runner_test.py [hunk 3]
```diff
@@ -17,7 +17,9 @@
 from __future__ import absolute_import
 from __future__ import print_function
 
+import functools
 import logging
+import io
```
Comment (udim) [inline sdks/python/apache_beam/runners/portability/fn_api_runner_test.py]: isort error from precommit:
```
04:33:03  import functools
04:33:03 +import io
04:33:03  import logging
04:33:03 -import io
04:33:03  import os
```

File: sdks/python/apache_beam/runners/portability/fn_api_runner_test.py (hunk 1 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/runners/portability/fn_api_runner_test.py]: `six` provides means to use some Python 2 syntactic idioms in Python 3. In Beam codebase we prefer to use idiomatic Python 3 as much as possible, and backport it to Python 2 using `python-future` if necessary.

File: sdks/python/apache_beam/runners/portability/fn_api_runner_test.py [hunk 4]
```diff
@@ -47,11 +49,31 @@
...
+  return wrapper
+
+
 class FnApiRunnerTest(unittest.TestCase):
 
   def create_pipeline(self):
     return beam.Pipeline(runner=fn_api_runner.FnApiRunner())
 
+  @capture_stderr
   def test_assert_that(self):
```
Comment (udim) [inline sdks/python/apache_beam/runners/portability/fn_api_runner_test.py]: This test is failing on Jenkins precommit.

**Comment by robertwb** [comment]
Run Python PreCommit

**Comment by robertwb** [comment]
I'm at a loss trying to figure out why swapping out stderr causes GRPC to hang. Removing that change, just doing the others for now. PTAL.

**Comment by udim** [comment]
Run Java PreCommit

**Comment by robertwb** [comment]
The failing java precommit looks like it's pre-existing and is unrelated to this change.

**Comment by robertwb** [comment]
Run Java PreCommit

**Comment by udim** [comment]
Run Java PreCommit

**Pull request closed without merging**